bundler/lib/bundler.rb

```diff
@@ -344,25 +344,7 @@ def clean_env
 
     # @return [Hash] Environment with all bundler-related variables removed
     def unbundled_env
-      env = original_env
-
-      if env.key?("BUNDLER_ORIG_MANPATH")
-        env["MANPATH"] = env["BUNDLER_ORIG_MANPATH"]
-      end
-
-      env.delete_if {|k, _| k[0, 7] == "BUNDLE_" }
-
-      if env.key?("RUBYOPT")
-        env["RUBYOPT"] = env["RUBYOPT"].sub "-rbundler/setup", ""
-      end
-
-      if env.key?("RUBYLIB")
-        rubylib = env["RUBYLIB"].split(File::PATH_SEPARATOR)
-        rubylib.delete(File.expand_path("..", __FILE__))
-        env["RUBYLIB"] = rubylib.join(File::PATH_SEPARATOR)
-      end
-
-      env
+      original_env.reject {|k, _| k[0, 7] == "BUNDLE_" }
```

Comment:
[Hash#reject](https://ruby-doc.org/core-2.7.1/Hash.html#method-i-reject) is used here in a similar way to `delete_if` was used before, the difference being reject does not change the env hash in place, but will just return a new hash containing entries that the block returned false for.